conda_smithy/schema.py

```diff
@@ -0,0 +1,1328 @@
+            swapfile_size="0GiB"
+        ),
+        description="Linux-specific settings for self-hosted runners",
+    )
+
+    settings_osx: AzureSelfHostedRunnerSettings = Field(
+        default_factory=lambda: AzureSelfHostedRunnerSettings(
+            pool={"vmImage": "macOS-11"}
+        ),
+        description="OSX-specific settings for self-hosted runners",
+    )
+
+    settings_win: AzureSelfHostedRunnerSettings = Field(
+        default_factory=lambda: AzureSelfHostedRunnerSettings(
+            pool={"vmImage": "windows-2022"},
+            variables={
+                "CONDA_BLD_PATH": "D:\\bld\\",
+                "UPLOAD_TEMP": "D:\\tmp",
+            },
+        ),
+        description="Windows-specific settings for self-hosted runners",
+    )
+
+    user_or_org: Optional[Union[str, Nullable]] = Field(
+        default=None,
+        description="The name of the GitHub user or organization, if passed with \
+        the GithubConfig provider, must comply with the value of the user_or_org field",
+        exclude=True,  # Will not be rendered in the model dump
+    )
+
+    store_build_artifacts: Optional[bool] = Field(
+        default=False,
+        description="Store the conda build_artifacts directory as an \
+        Azure pipeline artifact",
+    )
+
+    timeout_minutes: Optional[Union[int, Nullable]] = Field(
+        default=None,
+        description="The maximum amount of time (in minutes) that a \
+            job can run before it is automatically canceled",
+    )
+
+
+class GithubConfig(BaseModel):
+    user_or_org: Optional[str] = Field(
+        description="The name of the GitHub user or organization, \
+        if passed with the AzureConfig provider, must comply with the value of the user_or_org field",
+        default="conda-forge",
+    )
+    repo_name: Optional[str] = Field(
+        description="The name of the repository",
+        default="",
+    )
+    branch_name: Optional[str] = Field(
+        description="The name of the branch to execute on",
+        default="main",
+    )
+    tooling_branch_name: Optional[str] = Field(
+        description="The name of the branch to use for rerender+webservices \
+            github actions and conda-forge-ci-setup-feedstock references",
+        default="main",
+    )
+
+
+class GitHubActionsConfig(BaseModel):
+    artifact_retention_days: Optional[int] = Field(
+        description="The number of days to retain artifacts",
+        default=14,
+    )
+
+    max_parallel: Optional[Union[int, Nullable]] = Field(
+        description="The maximum number of jobs to run in parallel",
+        default=None,
+    )
+
+    self_hosted: Optional[bool] = Field(
+        description="Whether to use self-hosted runners",
+        default=False,
+    )
+
+    store_build_artifacts: Optional[bool] = Field(
+        description="Whether to store build artifacts",
+        default=False,
+    )
+
+
+class PlatformUniqueConfig(BaseModel):
+    enabled: bool = Field(
```

Comment:
I'm not sure whether we are using these keys anymore? e.g. [like this](https://github.com/conda-forge/libsimpleitk-feedstock/blob/983292ea3c2b439d30fbbbf617332cf0d43b21df/conda-forge.yml#L4-L7):

```yaml
linux_aarch64:
  enabled: false
linux_ppc64le:
  enabled: false
```

I thought that these days we just add them to the `build_platform` dict, and their presence in the dict is enough to enable them (the value would determine whether we emulate or cross-compile?).

Do you have any feedback @isuruf

Reply:
No, those are not user facing values. They are used internally in rendering.